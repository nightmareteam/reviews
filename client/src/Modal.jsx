import React from 'react';
//import styled from 'styled-components';
const styled = window.styled;

/**
 * This class is the template for all modals
 * @extends React.Component
 */
class Modal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  /**
   * Keeps the dialog box open on user click
   * @param {Object} e - an event object
   */
  onDialogClick(e) {
    // The overlay closes on click, but we need to make sure
    // if the click happens on the dialog, nothing happens
    e.stopPropagation();
  }

  render() {
    return (
      <ModalOverlay >
        <ModalContent onClick={this.props.onClose}>
          <ModalDialog onClick={this.onDialogClick}>
            {this.props.children}
          </ModalDialog>
        </ModalContent>
      </ModalOverlay>
    );
  }
}

// This is the dark background for the modal
const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  height: 100%;
  width: 100%;
  z-index: 1000;
  background-color: rgba(0, 0, 0, .65);
`;

const ModalContent = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 10000;
  overflow: auto;
  text-align: center;
  padding: 4px;
  cursor: pointer;
`;

const ModalDialog = styled.div`
  top: 100px;
  position: relative;
  background: #1b2738;
  outline: 0;
  padding: 30px;
  width: auto;
  display: inline-block;
  vertical-align: middle;
  box-sizing: border-box;
  max-width: auto;
  cursor: default;
  border-radius: 4px;
`;

export default Modal;