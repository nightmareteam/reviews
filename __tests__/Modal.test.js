/* eslint-disable camelcase */
import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../client/src/Modal.jsx';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

Enzyme.configure({ adapter: new Adapter() });

let wrapper;
let props;
let mockCallback;

const createProps = (mockCallback) => ({

});

beforeEach(() => {
  jest.resetModules();
  mockCallback = jest.fn();
  props = createProps(mockCallback);
  wrapper = shallow(<Modal onClick={mockCallback} /> );
});

describe('<Modal /> rendering', () => {
  it('should render elements', () => {
    expect(wrapper).toMatchSnapshot();
  });
});