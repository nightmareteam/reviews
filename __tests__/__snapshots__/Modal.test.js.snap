// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> rendering should render elements 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      onClick={[Function]}
    />
  </styled.div>
</styled.div>
`;
